#167733634 Integrate TravisCI with readme badge

danielufeli opened the pull request:
### What does this PR do?
Integrates TravisCI with readme badge

### Description of Task to be completed?
- Added and configured .travis.yml config file
- Added Travis badge to the readme file

### What are the relevant pivotal tracker stories?
#167733634

### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/48371156/63290856-3c20d100-c2ba-11e9-90e5-7a900d43c667.png)


[Finishes #167733634]

King-Success committed:
chore(docker): setup backend application to use do use docker
[Finishes #167733640]

King-Success committed:
chore(docker): setup backend application to use do use docker
[Finishes #167733640]

King-Success committed:
refactor(eslint): terminate files with a line break
[Finishes #167733640]

Baystef committed:
chore(test): setup unit testing

- installed mocha as the test runner
- installed chai and chai-http as the assertion library
- installed @babel/core, @babel/register to run mocha on es6
- configured babel with .babelrc and @babel/preset-env

[Delivers #167733635]

danielufeli committed:
chore(travisci):add travis yml

- Added and configured .travis.yml config file
- Added Travis badge to the readme file

[Finishes #167733634]

danielufeli committed:
chore(travisci):add travis yml

- Added and configured .travis.yml config file
- Added Travis badge to the readme file

[Finishes #167733634]

danielufeli committed:
chore(travisci):add travis yml

- Added and configured .travis.yml config file
- Added Travis badge to the readme file

[Finishes #167733634]

dchima committed:
chore(ci-badge): integrate houndCI with badge

This commit integrates HooundCI for ESLint reviewing
- added .hound.yml file for config
- edited readme to show houndci badge

[Delivers #167733633]

dchima committed:
chore(ci-badge): badge added

This commit integrates HooundCI for ESLint reviewing
- edited readme to show houndci badge

[Delivers #167733633]

dchima committed:
Revert "#167733633 Integrate HoundCI with badge attached to readme"

dchima reviewed (changes requested):
The travis CI is failing because you didn't add the following to the .travis.yml file:

**install:** 
     **- npm ci**
**script:**
    **- npm test**

The first will run travis in their site and the second will run whatever tests script we setup in the package.json. Maybe wait until the mocha test setup is approved before adding npm test portion

Baystef reviewed (approved):
Proof of Travis-ci spotted on the pull request and also badge is integrated properly

Baystef commented:
> The travis CI is failing because you didn't add the following to the .travis.yml file:
> 
> **install:**
> **- npm ci**
> **script:**
> **- npm test**
> 
> The first will run travis in their site and the second will run whatever tests script we setup in the package.json. Maybe wait until the mocha test setup is approved before adding npm test portion

chis0m committed:
install pg module, add env variables

dchima reviewed (approved):
It seems Travis still fails until a test script is passed (it does automatic testing). 
So I'm approving this PR.

daylay92 committed:
chore(setup): add ESlint configuration
[Finishes #167733632]

efeguono commented:
I'm going to wait until the PR that adds unit tests to the project is merged so that this can be fully tested.

efeguono commented on README.md:
```diff
@@ -1,4 +1,5 @@
-[![Reviewed by Hound](https://img.shields.io/badge/ESLint%20Reviewed%20by%20-HoundCI-d16ef5)](https://houndci.com)
```
Why was this removed?

danielufeli replied:
I made a PR to master before, but when I switched base to develop, it conflicted with Chima's code because he already merged to that branch, that's why I removed it.

efeguono replied:
That is a very wrong way to resolve merge conflicts. If this was a new feature, It is definitely sure that you would be introducing a bug to the application.

efeguono replied:
Always make enquiries (or be absolutely sure) before deleting any line of code. In any event, I have reverted the changes pertaining to the HoundCI integration. Be sure to rebase from develop and correctly resolve you merge conflicts

danielufeli replied:
Ok thank you noted @eguonoefe

efeguono commented on .travis.yml:
```diff
@@ -0,0 +1,7 @@
+services:
+  - postgresql
+notifications:
+  email: false
```
This requires a new line to be present at the end of the file

danielufeli replied:
ok @eguonoefe

Baystef committed:
fix(babelrc): add new line to end of .babelrc file

Baystef committed:
fix(chaiHttp): remove unnecessary chaiHttp

King-Success committed:
bug(dependency): install morgan as a dependency before attempting to use inside

chis0m committed:
Merge branch 'develop' into ch-database-setup-167733630

chis0m committed:
Merge branch 'develop' into ch-database-setup-167733630

danielufeli committed:
Merge branch 'ch-integrate-travisci-with-readme-badge-167733634' of https://github.com/andela/jubilee-backend into ch-integrate-travisci-with-readme-badge-167733634

King-Success committed:
refactor(Docker): update Dockerfile to run start:dev command and rename docker-compose services

efeguono committed:
Merge pull request #12 from andela/ch-setup-docker-for-backend-application-#167733640

#167733640 Setup backend application to use docker

efeguono committed:
Merge pull request #8 from andela/ch-database-setup-167733630

#167733630 Configure BareFoot Nomad to use PostgreSQL as DB Engine

chis0m committed:
chore(sequelize): setup sequelize

chis0m committed:
add new line to file

Baystef committed:
fix(vscode): delete unnecessary config file

chis0m committed:
update sequelize default env variable

chis0m committed:
update sequelize default env variables

efeguono committed:
Merge pull request #15 from andela/ch-setup-sequelize-167733631

#167733631: Setup Sequelize

Baystef committed:
fix(conflict): resolve package.json conflicts

efeguono committed:
Merge pull request #3 from andela/ch-setup-unit-testing

#167733635 Setup Test environment for the project

dchima committed:
bg(houndCI): adds fix to houndCI configuration

This commit changes .hound.yml file to point to eslint congif file
[FInished #167733633]

efeguono committed:
Merge pull request #16 from andela/bg-houndCI-fix-167733633

#167733633 Fix HoundCI Integration

danielufeli committed:
did npm install to resolve issue with travis

danielufeli committed:
Merge branch 'ch-integrate-travisci-with-readme-badge-167733634' of https://github.com/andela/jubilee-backend into ch-integrate-travisci-with-readme-badge-167733634

danielufeli committed:
modified package-lock.json file to snyc with package.json

dchima reviewed (approved):
Travis is passing on initial build, and badge shows fine on readme. Maybe update the screenshot on the PR description to show passing.

danielufeli commented:
> Travis is passing on initial build, and badge shows fine on readme. Maybe update the screenshot on the PR description to show passing.

ok updated screenshot

Segun-Ogundipe reviewed (approved):
This is great!!! I confirmed Travisci has been properly integrated

daylay92 reviewed (commented):
A new line should be added at the end of every file including this one

daylay92 commented on .travis.yml:
```diff
@@ -0,0 +1,8 @@
+  - postgresql
+notifications:
+  email: false
+  
```
A new line should be added at the end of every file

danielufeli replied:
I added a new line. @daylay92

daylay92 replied:
okay

daylay92 replied:
Since the package.json file has been updated alongside the test script , I think you can now update the .travis.yml file with the install and script properties like so:
*install :* 
**npm install**
*script :*
**npm test**
